### ISSUE TITLE
[BUG] Expand Selection and Shrink Selection do not work on VSCode

### ISSUE THREAD

**Issue by ATheCoder** [RedBearAK/toshy#194]
```
Bare metal or virtual machine: 
(If in VM, which VM software): None

(Try running 'toshy-env' in a terminal.)

(EV) Toshy env module sees this environment:
		DISTRO_NAME     = 'endeavouros'
		DISTRO_VER      = 'notfound'
		VARIANT_ID      = 'notfound'
		SESSION_TYPE    = 'wayland'
		DESKTOP_ENV     = 'kde'
		DE_MAJ_VER      = '6'

Any DE(s) tried with Wayland?: 

Keyboard type (IBM, Chromebook, Windows, Apple): 
Keyboard device name (try 'toshy-devices'): 

/dev/input/event22   Keychron Keychron K2                usb-0000:00:14.0-12/input1
/dev/input/event21   Keychron Keychron K2                usb-0000:00:14.0-12/input0

```
Problem observed:  

The `Expand Selection` and `Shrink Selection` Keyboard Shortcuts dont work in vscode.

**Comment by RedBearAK** [comment]
@ATheCoder 

Does the diagnostic shortcut work? See the README. 

What are the actual Mac-like shortcuts, and what physical keys make them work if Toshy is disabled?

**Comment by ATheCoder** [comment]
@RedBearAK Thanks for the fast response. :)

The actual shortcuts configured in VSCode are `Shift + Alt + Left Arrow` and `Shift + Alt + Right Arrow`.

I am using Left Control + Shift + Right/Left Arrow on my Keyboard.

This also didn't work in Kinto and I had to add the following inside Kinto to make it work:

```
   K("LSuper-LShift-RIGHT"): K("LShift-Alt-Right"),
    K("LSuper-LShift-LEFT"): K("LShift-Alt-Left"),
```

But adding the same inside the `~/.config/toshy/toshy_config.py` but with `C` instead of `K` doesn't work.

**Comment by ATheCoder** [comment]
Btw, everything else seems fine. I am sure that Toshy is working fine its just that this specific shortcut doesn't work.

**Comment by RedBearAK** [comment]
I am pretty sure you should be using the key at the physical position of the Apple Option/Alt key for this. It should work, I use it frequently. 

If that’s not the problem I’ll have to take a closer look at it in the morning. 

It may work on physical Ctrl if you use “Super” instead of “LSuper”. But I don’t think those are the correct physical keys to “Work like a ’Tosh”. 

Also, changes you make outside the editable “slices” will be kept in a backup but won’t be restored if you ever reinstall Toshy.

**Comment by RedBearAK** [comment]
@ATheCoder 

Looks like I was misunderstanding the exact shortcuts you were referring to, and what they were supposed to be doing. 

I looked at a VSCode installed on Big Sur recently, with no customizations. There are two different sets of shortcuts assigned to these functions: 

```sh
editor.action.smartSelect.expand
editor.action.smartSelect.shrink
```

The two sets of shortcuts for "Expand Selection" are: 

```sh
Ctrl+Shift+Right
Ctrl+Shift+Cmd+Right
```

And for the "Shrink Selection" its: 

```sh
Ctrl+Shift+Left
Ctrl+Shift+Cmd+Left
```

I placed this in the "VSCodes" keymap as a test, and it worked for me: 

```py
keymap("VSCodes", {

    C("Shift-Super-Left"):      C("Shift-Alt-Left"),
    C("Shift-Super-Right"):     C("Shift-Alt-Right"),

    # C("Super-Space"):           C("LC-Space"),                  # Basic code completion (conflicts with input switching)

    # Override the global Cmd+Dot (Escape/cancel) shortcut for QuickFix in VSCode(s)
    C("RC-Dot"):                C("RC-Dot"),                    # QuickFix, overriding global shortcut
…
```

The most likely explanation for why you couldn't get it to work is the unnecessary usage of the "L" left variant of the key names in both the left (input) and right (output) combos. That would be important in some cases, but shouldn't have any relevance here on either side. 

I'll be adding both of these sets of shortcuts to the "VSCodes" keymap, something like this. 

```py
    C("Shift-Super-Right"):     C("Shift-Alt-Right"),           # Expand Selection (increase logical scope of smart selection)
    C("Shift-Super-Left"):      C("Shift-Alt-Left"),            # Shrink Selection (reduce logical scope of smart selection)

    C("Shift-Super-RC-Right"):  C("Shift-Alt-Right"),           # Expand Selection (increase logical scope of smart selection) [alt]
    C("Shift-Super-RC-Left"):   C("Shift-Alt-Left"),            # Shrink Selection (reduce logical scope of smart selection) [alt]
```

This has been merged into the main branch now. 

The way it works with the Toshy installer is any edits you make inside the "slice marks" will be preserved after a reinstall, but anything outside those marks will disappear (but will exist in a timestamped backup folder). So if this is the only thing you tried to get working by making edits outside the special marks, you should be safe to download the latest zip and reinstall Toshy to get these new remaps in the VSCodes keymap. 

Let me know how it goes. Apologies for the confusion on my part regarding the correct keys to use for these shortcuts that I was somehow not familiar with.

**Comment by RedBearAK** [comment]
Closing as resolved by PR #199.
